src/python_testing/TC_CNET_4_12.py

```diff
@@ -0,0 +1,355 @@
+            TestStep(2, 'TH reads Networks attribute from the DUT and saves the number of entries as NumNetworks'),
+            TestStep(3, 'TH saves the index of the Networks list entry from step 2 as Userth_netidx'),
+            TestStep(4, '''TH sends RemoveNetwork Command to the DUT 
+                     with NetworkID field set to th_xpan 
+                     and Breadcrumb field set to 1'''),
+            TestStep(5, '''TH sends AddOrUpdateThreadNetwork command to the DUT 
+                     with operational dataset field set to PIXIT.CNET.THREAD_2ND_OPERATIONALDATASET 
+                     and Breadcrumb field set to 1'''),
+            TestStep(6, 'TH reads Networks attribute from the DUT'),
+            TestStep(7, '''TH sends ConnectNetwork command to the DUT 
+                     with NetworkID field set to PIXIT.CNET.THREAD_2ND_OPERATIONALDATASET 
+                     and Breadcrumb field set to 2'''),
+            TestStep(8, 'TH discovers and connects to DUT on the PIXIT.CNET.THREAD_2ND_OPERATIONALDATASET operational network'),
+            TestStep(9, 'TH reads Breadcrumb attribute from the General Commissioning cluster of the DUT'),
+            TestStep(10, 'TH sends ArmFailSafe command to the DUT with ExpiryLengthSeconds set to 0'),
+            TestStep(11, 'TH ensures it can communicate on PIXIT.CNET.THREAD_1ST_OPERATIONALDATASET'),
+            TestStep(12, 'TH discovers and connects to DUT on the PIXIT.CNET.THREAD_1ST_OPERATIONALDATASET operational network'),
+            TestStep(13, 'TH sends ArmFailSafe command to the DUT with ExpiryLengthSeconds set to 900'),
+            TestStep(14, '''TH sends RemoveNetwork Command to the DUT with NetworkID field set to th_xpan 
+                     and Breadcrumb field set to 1'''),
+            TestStep(15, '''TH sends AddOrUpdateThreadNetwork command to the DUT 
+                     with the OperationalDataset field set to PIXIT.CNET.THREAD_2ND_OPERATIONALDATASET 
+                     and Breadcrumb field set to 1'''),
+            TestStep(16, '''TH sends ConnectNetwork command to the DUT 
+                     with NetworkID field set to the extended PAN ID of PIXIT.CNET.THREAD_2ND_OPERATIONALDATASET 
+                     and Breadcrumb field set to 3'''),
+            TestStep(17, 'TH discovers and connects to DUT on the PIXIT.CNET.THREAD_2ND_OPERATIONALDATASET operational network'),
+            TestStep(18, 'TH reads Breadcrumb attribute from the General Commissioning cluster of the DUT'),
+            TestStep(19, 'TH sends the CommissioningComplete command to the DUT'),
+            TestStep(20, 'TH reads Networks attribute from the DUT')
+        ]
+        return steps
+
+    @async_test_body
+    async def test_TC_CNET_4_12(self):
+
+        if self.is_pics_sdk_ci_only:
+            logger.info('Test is not running in CI.')
+            self.skip_all_remaining_steps('precondition-1')
+            return
+
+        # Pre-Conditions
```

Comment:
Let leave only _two_ preconditions: `precondition-1` and `precondition-2`. Please make sure that test raises error if any of the following PIXIT values had not been provided:
```
PIXIT.CNET.ENDPOINT_THREAD
PIXIT.CNET.THREAD_1ST_OPERATIONALDATASET
PIXIT.CNET.THREAD_2ND_OPERATIONALDATASET
```

Please create a docstring for test class that mentions that all the PIXITs above are mandatory.